edx_solutions_projects/management/commands/purge_deleted_groups_cohorts.py

```diff
@@ -0,0 +1,46 @@
+from django.core.management.base import BaseCommand
+
+from edx_solutions_api_integration.courseware_access import get_course_key
+from edx_solutions_projects.models import Project
+
+from openedx.core.djangoapps.course_groups.models import CourseUserGroup
+
+
+class Command(BaseCommand):
+    help = "Cohort deletion of course deleted groups"
+
+    def add_arguments(self, parser):
+        parser.add_argument('remove', nargs=1, type=str, help='Remove cohort')
+        parser.add_argument('course', nargs='+', type=str, help='Cohort deletion course')
+
+    def handle(self, *args, **options):
+        try:
+            op = options['remove'][0]
+            courses = options['course']
+            if op.lower() != 'remove':
+                raise Exception("Only 'remove' operations supported")
+
+            course_keys = [get_course_key(course_id) for course_id in courses]
+            if None in course_keys:
+                raise Exception("At least one course failed to create course key")
+
+            # Default cohort must exclude e.g default_cohort
+            exclude_cohorts = [CourseUserGroup.default_cohort_name]
+            projects = Project.objects.filter(course_id__in=courses).prefetch_related('workgroups')
```

Comment:
Why don't we query on workgroups? `Workgroup.objects.filter(project__course_id__in=courses)`

Reply:
Addressed